clang/test/SemaCXX/cxx20-ctad-type-alias.cpp

```diff
@@ -190,13 +196,15 @@ template <class T> struct Foo { Foo(T); };
 
 template<class V> using AFoo = Foo<V *>;
 template<typename> concept False = false;
-template<False W> using BFoo = AFoo<W>;
+template<False W>
+using BFoo = AFoo<W>; // expected-note {{candidate template ignored: constraints not satisfied [with V = int]}} \
+                      // expected-note {{because '__is_deducible(BFoo, Foo<int *>)' evaluated to false}} \
```

Comment:
I think we want to add code in this PR to improve the diagnostic

Reply:
Do you think this is a blocker for landing this PR?  The current state is acceptable to me, although not ideal. I plan to send out a follow-up patch to address it.

Reply:
I'd prefer to do it in a follow up (this patch is quite large already). Added a FIXME.